### PR TITLE
watchdog:add vm_destory to watchdog in case it affects subsequent test

### PR DESCRIPTION
**Pull request by MiriamDeng**
ID:2173526
Signed-off-by: MiriamDeng <mdeng@redhat.com>

### CODE REVIEW

**Comment by MiriamDeng** [comment]
(1/2) Host_RHEL.m8.u8.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.8.0.ppc64le.io-github-autotest-qemu.watchdog.migration_when_wdt_timeout.reboot_guest.i6300esb: STARTED
 (1/2) Host_RHEL.m8.u8.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.8.0.ppc64le.io-github-autotest-qemu.watchdog.migration_when_wdt_timeout.reboot_guest.i6300esb: PASS (215.10 s)
 (2/2) Host_RHEL.m8.u8.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.8.0.ppc64le.io-github-autotest-qemu.watchdog.migration_when_wdt_timeout.pause.i6300esb: STARTED
 (2/2) Host_RHEL.m8.u8.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.8.0.ppc64le.io-github-autotest-qemu.watchdog.migration_when_wdt_timeout.pause.i6300esb: PASS (153.24 s)

**Comment by MiriamDeng** [comment]
@PaulYuuu could you please give a hand to review it  ? Thanks

**Comment by MiriamDeng** [comment]
@yiqianwei can you give a hand to reivew it ? Thanks a lot.

**Comment by yiqianwei** [comment]
@MiriamDeng , LGTM 

for x86_64:
 (1/2) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.watchdog.migration_when_wdt_timeout.reboot_guest.i6300esb.q35: STARTED
 (1/2) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.watchdog.migration_when_wdt_timeout.reboot_guest.i6300esb.q35: PASS (210.61 s)
 (2/2) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.watchdog.migration_when_wdt_timeout.pause.i6300esb.q35: STARTED
 (2/2) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.watchdog.migration_when_wdt_timeout.pause.i6300esb.q35: PASS (172.27 s)

File: qemu/tests/watchdog.py [hunk 1]
```diff
@@ -458,3 +458,6 @@ def heartbeat_test():
     else:
         test.error("Oops test %s doesn't exist, have a check please."
                    % test_type)
+    for vm in env.get_all_vms():
+        if vm.is_alive():
+            vm.destroy()
```
Comment (PaulYuuu) [inline qemu/tests/watchdog.py:463]: This step will not run when the test fails. Suggest using a try block to handle it.
```python
    if (test_type in locals()):
        test_running = locals()[test_type]
        try:
            test_running()
        finally:
            vm.destroy()
    else:
        test.error("Oops test %s doesn't exist, have a check please."
                   % test_type)
```

Comment (MiriamDeng) [reply qemu/tests/watchdog.py]: Thanks, updated ! Please help to review it again !

**Review by PaulYuuu — APPROVED** [review]

**Comment by PaulYuuu** [comment]
Hello @yiqianwei, could you help to test it again, thanks.

**Comment by yiqianwei** [comment]
@PaulYuuu , It's okay.

Test results:

 (1/2) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.watchdog.migration_when_wdt_timeout.reboot_guest.i6300esb.q35: PASS (177.37 s)

 (2/2) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.watchdog.migration_when_wdt_timeout.pause.i6300esb.q35: PASS (138.22 s)

**Pull request merged**